src/sentry/utils/services.py

```diff
@@ -58,7 +60,13 @@ def setup(self) -> None:
         """
 
 
-class LazyServiceWrapper(LazyObject):
+if TYPE_CHECKING:
+    Proxied = TypeVar("Proxied", bound=Service)
+else:
+    Proxied = object
+
+
+class LazyServiceWrapper(LazyObject, Proxied):
```

Comment:
does this work? I didn't know you could inherit from a TypeVar

Reply:
completely broken at runtime, so if we set `typing.TYPE_CHECKING` to true (which e.g. happens in sphinx-autodoc-typehints), everything blows up

mypy/pyright are fine though